packages/core/src/simulate.ts

```diff
@@ -0,0 +1,45 @@
+import type {
+  AnyActorContext,
+  AnyActorLogic,
+  EventFromLogic,
+  InternalStateFrom
+} from '.';
+import { createEmptyActor } from './actors';
+
+function createMockActorContext(): AnyActorContext {
+  const emptyActor = createEmptyActor();
+  return {
+    self: emptyActor,
+    logger: console.log,
+    id: 'root_test',
+    sessionId: Math.random().toString(32).slice(2),
+    defer: () => {},
+    system: emptyActor,
+    stopChild: () => {}
+  };
+}
+
+export function simulate<T extends AnyActorLogic>(
+  actorLogic: T,
+  options: { input?: any } = {}
+) {
+  const dummyActorContext = createMockActorContext();
+
+  const sim = {
+    transition: (
+      state: InternalStateFrom<T> = actorLogic.getInitialState(
+        dummyActorContext,
+        options.input
+      ),
+      event: EventFromLogic<T>
+    ): InternalStateFrom<T> => {
+      return actorLogic.transition(state, event, dummyActorContext);
```

Comment:
Through `@xstate/test` we are kinda branching off different potential paths while simulating. `actorContext` provides access to a mutable resource that is being shared among many "timelines" created by branching those paths etc. This sounds a little bit disastrous... but I'm also not quite sure yet what we should do about it here etc. It would be easier if people just... wouldnt use anything from this context when using `@xstate/test` 😅 but that's hard to forbid as well.

For the time being, I'm just mentioning this to raise awareness that this is a little bit dangerous - but I don't have any proposals at all as to what we should do here instead.